app/forms/developer_search_form.rb

```diff
@@ -9,20 +9,14 @@ class DeveloperSearchForm
   attribute :programming_language_id, Integer
 
   def search
-    query = Developer.all
-    if language_id.present?
-      query = query.includes(:developer_languages)
-                   .joins(:developer_languages)
-                   .where(developer_languages: { language_id: language_id })
-    end
+    query = Developer.includes(:developer_languages,
+                               :languages,
+                               :developer_programming_languages,
+                               :programming_languages)
```

Comment:
Looks way better!! 👍 
That's what I want.
But I wonder if all of the includes are necessary...

Reply:
What if you remove `developer_languages` and `developer_programming_languages`?

Reply:
Thanks, I tried remove, but when Searching with param 'language' or 'programming_language' has alert such as https://gyazo.com/f1873ba71793bcd9305d8fb204c34e90

Reply:
Did you actually see inefficient queries?
bullet doesn't work perfectly.
So when you see alerts by bullet, make suer if there're inefficient queries actually.
If not, add whitelist as follows.
https://github.com/flyerhzm/bullet#whitelist

Reply:
Thanks. I checked again and I fixed as your guide 🌝